utils.nix

```diff
@@ -31,4 +31,6 @@ in
   # systemd recommends multi-user.target over default.target.
   # https://www.freedesktop.org/software/systemd/man/latest/systemd.special.html#default.target
   defaultTarget = if isUserSystemd then "default.target" else "multi-user.target";
+
+  podmanPackage = config.virtualisation.podman.package or osConfig.virtualisation.podman.package or pkgs.podman;
```

Comment:
let's just take a `podmanPackage` argument in this file and defer this selection to nixos / hm module.